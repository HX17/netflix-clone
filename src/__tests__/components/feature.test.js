import React from "react";
import { render } from "@testing-library/react";
import Feature from "../../components/feature";

describe("<Feature />", () => {
  it("renders the <Feature /> with populated data", () => {
    const { container, getByText } = render(
      <Feature>
        <Feature.Title>Unlimited movies, TV shows and more.</Feature.Title>
        <Feature.Subtitle>Watch anywhere. Cancel at any time.</Feature.Subtitle>
      </Feature>
    );

    expect(getByText("Unlimited movies, TV shows and more.")).toBeTruthy();
    expect(getByText("Watch anywhere. Cancel at any time.")).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders the <Feature /> with just a title", () => {
    const { container, getByText, queryByText } = render(
      <Feature>
        <Feature.Title>Unlimited movies, TV shows and more.</Feature.Title>
      </Feature>
    );

    expect(getByText("Unlimited movies, TV shows and more.")).toBeTruthy();
    expect(queryByText("Watch anywhere. Cancel at any time.")).toBeFalsy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
