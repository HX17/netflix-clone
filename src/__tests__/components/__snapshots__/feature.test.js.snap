// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Feature /> renders the <Feature /> with just a title 1`] = `
<div
  class="feature__Container-sc-19zs3jp-0 hPnhgU"
>
  <h1
    class="feature__Title-sc-19zs3jp-1 lhpEBX"
  >
    Unlimited movies, TV shows and more.
  </h1>
</div>
`;

exports[`<Feature /> renders the <Feature /> with populated data 1`] = `
<div
  class="feature__Container-sc-19zs3jp-0 hPnhgU"
>
  <h1
    class="feature__Title-sc-19zs3jp-1 lhpEBX"
  >
    Unlimited movies, TV shows and more.
  </h1>
  <h2
    class="feature__Subtitle-sc-19zs3jp-2 hWNCzI"
  >
    Watch anywhere. Cancel at any time.
  </h2>
</div>
`;
